choicemodels/tools/mergedchoicetable.py

```diff
@@ -121,12 +123,25 @@ def __init__(self, observations, alternatives, chosen_alternatives=None,
         # TO DO - check that dfs have unique indexes
         # TO DO - check that chosen_alternatives correspond correctly to other dfs
         # TO DO - same with weights (could join onto other tables and then split off)
-        # TO DO - check for overlapping column names
         
         # Normalize chosen_alternatives to a pd.Series
         if (chosen_alternatives is not None) & isinstance(chosen_alternatives, str):
-            chosen_alternatives = observations[chosen_alternatives]
+            chosen_alternatives = observations[chosen_alternatives].copy()
             observations = observations.drop(chosen_alternatives.name, axis='columns')
+            chosen_alternatives.name = '_' + alternatives.index.name  # avoids conflicts
+        
+        # Check for duplicate column names
+        obs_cols = list(observations.columns) + list(observations.index.names)
+        alt_cols = list(alternatives.columns) + list(alternatives.index.names)
+        dupes = [c for c in obs_cols if c in alt_cols]
```

Comment:
Why is this using lists (`O(a*o)`) instead of sets (`O(min(a,o))`)? like:

```
        obs_cols = set(observations.columns) + set(observations.index.names)
        alt_cols = set(alternatives.columns) + set(alternatives.index.names)
        dupes = obs_cols & in alt_cols
```

Reply:
@Eh2406 Thanks, that's definitely better! Unfortunately i just merged this PR, but i'll update this in the next one

Reply:
https://github.com/UDST/choicemodels/commit/f8173a9cc5567acbaa5abe8e1bb6f3a11db890fa